plugins/amazonq/codetransform/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codemodernizer/controller/CodeTransformChatController.kt

```diff
@@ -359,6 +382,146 @@ class CodeTransformChatController(
+        codeTransformChatHelper.addNewMessage(buildTransformFindingLocalAlternativeDependencyChatContent())
+        val createReportResult = codeModernizerManager.createDependencyReport(hilDownloadArtifact)
+        if (createReportResult == MavenDependencyReportCommandsResult.Cancelled) {
+            hilTryResumeAfterError(message("codemodernizer.chat.message.hil.error.cancel_dependency_search"))
+            return
+        } else if (createReportResult == MavenDependencyReportCommandsResult.Failure) {
+            hilTryResumeAfterError(message("codemodernizer.chat.message.hil.error.no_other_versions_found", hilDownloadArtifact.manifest.pomArtifactId))
+            return
+        }
+
+        val dependency = codeModernizerManager.findAvailableVersionForDependency(
+            hilDownloadArtifact.manifest.pomGroupId,
+            hilDownloadArtifact.manifest.pomArtifactId
+        )
+
+        if (dependency == null || (dependency.majors.isNullOrEmpty() && dependency.minors.isNullOrEmpty() && dependency.incrementals.isNullOrEmpty())) {
+            hilTryResumeAfterError(message("codemodernizer.chat.message.hil.error.no_other_versions_found", hilDownloadArtifact.manifest.pomArtifactId))
+            return
+        }
+        codeTransformChatHelper.updateLastPendingMessage(buildTransformAwaitUserInputChatContent(dependency))
+        runInEdt {
+            codeModernizerManager.getBottomToolWindow().show()
+        }
+    }
+
+    override suspend fun processConfirmHilSelection(message: IncomingCodeTransformMessage.ConfirmHilSelection) {
+        if (!checkForAuth(message.tabId)) {
+            return
+        }
+
+        val selectedVersion = message.version
+        val artifact = codeModernizerManager.getCurrentHilArtifact() as CodeTransformHilDownloadArtifact
+
+        codeTransformChatHelper.run {
+            addNewMessage(buildUserHilSelection(artifact.manifest.pomArtifactId, artifact.manifest.sourcePomVersion, selectedVersion))
+
+            addNewMessage(buildCompileHilAlternativeVersionContent())
+        }
+
+        val copyDependencyResult = codeModernizerManager.copyDependencyForHil(selectedVersion)
+        if (copyDependencyResult == MavenCopyCommandsResult.Failure) {
+            hilTryResumeAfterError(message("codemodernizer.chat.message.hil.error.cannot_upload"))
+            return
+        } else if (copyDependencyResult == MavenCopyCommandsResult.Cancelled) {
+            hilTryResumeAfterError(message("codemodernizer.chat.message.hil.error.cancel_upload"))
+            return
+        }
+
+        try {
+            codeModernizerManager.tryResumeWithAlternativeVersion(selectedVersion)
+
+            telemetry.logHil(
+                CodeModernizerSessionState.getInstance(context.project).currentJobId?.id as String,
+                HilTelemetryMetaData(
+                    versionSelected = selectedVersion,
+                    reason = "User selected version",
+                    result = "SUCCESS",
```

Comment:
NIT: Should be telemetry enum for success and Failure

Reply:
Added an enum for it.